src/Endpoints/AbstractEndpoint.php

```diff
@@ -7,17 +7,16 @@
  */
 namespace Qdrant\Endpoints;
 
-use GuzzleHttp\Psr7\HttpFactory;
-use GuzzleHttp\Psr7\Query;
-use Psr\Http\Message\RequestInterface;
+use Qdrant\ClientInterface;
 use Qdrant\Exception\InvalidArgumentException;
-use Qdrant\Http\HttpClientInterface;
 
 abstract class AbstractEndpoint
 {
     protected ?string $collectionName = null;
 
-    public function __construct(protected HttpClientInterface $client)
+    use HttpFactoryTrait;
```

Comment:
Why we need this as trait?

Reply:
Both ways work for me. We can either copy all the methods into AbstractEndpoint, which we already have in HttpFactoryTrait, or we can keep the specific methods separate. However, I prefer keeping them separate. I am also not satisfied with having setCollection() and getCollection() methods in AbstractEndpoint, as not all endpoints use them.